### PR TITLE
feat: `toHaveDisplayValue` matcher

### LINKED ISSUE #1456: feature: implement `toHaveDisplayValue` Jest matcher
## Describe the Feature

`toHaveDisplayValue` is a [Jest DOM matcher](https://github.com/testing-library/jest-dom#tohavedisplayvalue) that asserts the value visible to the user in text inputs. 

Adapting it to our case it would assert the value in `TextInput` host elements, to match given value.

## Possible Implementations
Proposed API:

```ts
export function toHaveDisplayValue(
  this: jest.MatcherContext,
  element: ReactTestInstance,
  valueToMatch: TextMatch,
  options?: TextMatchOptions
)
```

The matcher should:
1) Validate that it is invoked on host TextInput element
2) Get the current value as `props.value ?? props.defaultValue`
3) Compare it with passed `valueToMatch`.

Each matcher should have a fairly comprehensive test suite.

## Links

* `toHaveDisplayValue` matcher code from Jest DOM: https://github.com/testing-library/jest-dom/blob/main/src/to-have-display-value.js

## Related Issues
#1454

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Implement `toHaveDisplayValueMatcher()`
Resolves #1456 

### Test plan

Added relevant tests.

### CODE REVIEW

File: src/matchers/__tests__/to-have-display-value.tsx
[the remark this answers is not in the document]
Comment (jaworek) [reply src/matchers/__tests__/to-have-display-value.tsx:1]: done :D

**Review by mdjastrzebski — COMMENTED** [review]
LGTM

**Comment by mdjastrzebski** [comment]
I've added some additional test and extracted some common TextInput functions used across RNTL.

**Comment by mdjastrzebski** [comment]
🎉 Release in version: [v12.4.0](https://github.com/callstack/react-native-testing-library/releases/tag/v12.4.0)

**Pull request merged**